main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1420,6 +1420,12 @@ object Resolver {
               }
           }
 
+        case NamedAst.Expr.InvokeConstructor2(clazzName, exps, loc) =>
+          val esVal = traverse(exps)(visitExp(_, env0))
+          mapN(esVal) {
+            es => ResolvedAst.Expr.InvokeConstructor2(Class.forName(clazzName.name), es, loc)
```

Comment:
Here we will need some more complex code that uses the import environment.

Maybe @mlutze can sketch the solution.